packages/template-ui/src/views/Home.vue

```diff
@@ -129,7 +130,19 @@ export default {
       else {
         // FIXME add API to query random content nodes
         console.debug(`Fetch ${this.carouselSlideNumber} random content nodes`);
-        this.loadingCarouselNodes = false;
+
+        window.kolibri.getContentByFilter({
+          extra: {
+            channel_id: this.channel.id,
```

Comment:
All APIs should be constrained to the current channel, to avoid leaking outside the sandbox. So there shouldn't be a need to pass the channel ID.

Reply:
This is something that's not implemented on the API, right now you only can pass the parent, but the parent could be any ID and if you don't pass the parent, there's no filter. If the main idea is to limit this API to the current channel, we can add the `channel_id` filter to all the API calls.

Reply:
Yeah :+1: we should filter all API calls by the current channel.